test/functional/publishing/batch_publishing_test.rb

```diff
@@ -192,6 +192,53 @@ def setup
     assert_select 'a[href=?]', data_file_path(not_requested_df), count: 0
   end
 
+  test 'authorization for cancel_publishing_request' do
+    gatekeeper = FactoryBot.create(:asset_gatekeeper)
+    gatekept_project = gatekeeper.projects.first
+    a_person = FactoryBot.create(:person)
+    a_person.add_to_project_and_institution(gatekept_project, FactoryBot.create(:institution))
+    df = FactoryBot.create(:data_file, contributor: a_person, projects: [gatekept_project])
+    df.resource_publish_logs.create(publish_state: ResourcePublishLog::WAITING_FOR_APPROVAL, user: a_person.user)
+
+    login_as(gatekeeper)
+    get :cancel_publishing_request, params: { id: a_person,
+                                              asset_id: df.id,
+                                              asset_class: df.class }
+    assert_redirected_to :root
+    assert_not_nil flash[:error]
+
+    login_as(a_person)
```

Comment:
would this fail if a gatekeeper, but from a different project to the data file ?

Reply:
I think it needs an additional test scenario for this

Reply:
I ended up with 4 scenarios in the tests here: [d54c00f](https://github.com/seek4science/seek/pull/1484/commits/d54c00f191273ad8c8ccb8c9ac8839824f7dc2b9).

I went for anyone with manage rights can cancel the request, although at the minute the only one that can "click" a button to cancel is the one who created the request, because only they can see the item in their waiting_approval_items view...

Through the API, though, anyone with manage rights should be able to cancel.